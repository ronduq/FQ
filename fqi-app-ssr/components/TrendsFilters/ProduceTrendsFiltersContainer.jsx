import { connect } from 'react-redux'
import { bindActionCreators } from 'redux';
import TrendsFilters from './TrendsFilters';

import { setCompareByThunk } from '../../actions/produceTrends';
import { selectAllRetailersThunk, updateRetailerTrendsThunk } from '../../actions/retailers';
import { getFiltersContent } from '../../selectors/index'

const mapStateToProps = state => {
  return {
    filtersContent: getFiltersContent(state),
    title: getFiltersContent(state).retailers,
  }
}

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      updateSelection: (produceId, check) => updateRetailerTrendsThunk(produceId, check),
      selectAll: (status) => selectAllRetailersThunk(status, true),
      setCompareBy: (value) => setCompareByThunk(value),
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TrendsFilters);
