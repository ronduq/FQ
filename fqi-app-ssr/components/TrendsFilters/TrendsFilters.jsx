import React from 'react';
import PropTypes from 'prop-types';

import CompareByFilter from '../CompareByFilter/CompareByFilter';
import CollapseableFilter from '../CollapseableFilter/CollapseableFilter';


import s from './TrendsFilters.scss';

const TrendsFilters = ({
  data,
  selectAll,
  updateSelection,
  compareBy,
  setCompareBy,
  filtersContent,
  title,
}) => {
  return (
    <>
      <div className={s.filterTitle}>{filtersContent.main_title}</div>
        <CompareByFilter sortRanks={setCompareBy} selected={compareBy} showPerception={false} content={filtersContent}/>
        <div className={s.filterContainer}>
          <CollapseableFilter 
            data={data}
            updateItem={updateSelection}
            selectAllItems={selectAll}
            title={title}
          />
        </div>
    </>
  );
};

TrendsFilters.propTypes = {
  data: PropTypes.shape(),
  selectAll: PropTypes.func,
  updateSelection: PropTypes.func,
  compareBy: PropTypes.string,
  setCompareBy: PropTypes.func,
};

export default TrendsFilters;
