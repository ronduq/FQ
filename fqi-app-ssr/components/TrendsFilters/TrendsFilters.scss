@import '../../style/settings/colours';
@import '../../style/settings/fonts';
@import '../../style/settings/breakpoints';

.filterContainer {
  background: white;
  border: 1px solid $colour-grey-medium;
  padding: 15px 15px;
}

.filterTitle {
  @include font-family('proxima-regular');
  @include font-size('medium');
  background: white;
  border: 1px solid $colour-grey-medium;
  padding:  5px 15px;
  box-shadow: 0 4px 2px -2px rgba(126,126,126, 0.4) ;

  @include respond-to('tablet', min-width) {
    @include font-size('medium-20');
  }
}
