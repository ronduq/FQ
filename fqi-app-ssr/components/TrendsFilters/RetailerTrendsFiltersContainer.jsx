import { connect } from 'react-redux'
import { bindActionCreators } from 'redux';
import TrendsFilters from './TrendsFilters';

import { selectAllProducesThunk, updateProduceTrendsThunk } from '../../actions/producesActions';
import { setCompareByThunk } from '../../actions/retailerTrends';
import { getFiltersContent } from '../../selectors/index'

const mapStateToProps = state => {
  return {
    filtersContent: getFiltersContent(state),
    title: getFiltersContent(state).produce,
  }
}
const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      updateSelection: (produceId, check, parentCode, isParent) => updateProduceTrendsThunk(produceId, check, parentCode, isParent),
      selectAll: (status) => selectAllProducesThunk(status, true),
      setCompareBy: (value) => setCompareByThunk(value),
    },
    dispatch
  );

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TrendsFilters);
