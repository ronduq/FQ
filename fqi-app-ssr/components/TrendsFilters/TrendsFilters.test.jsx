/* eslint-env jest */

import React from 'react'
import { Provider } from 'react-redux';
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import {
  initialProducesState, 
} from '../../reducers/initialState'

import TrendsFilters from './TrendsFilters';

jest.mock('../../hocs/Layout', () => ({
  __esModule: true,
  default: (e) => e,
}));

const fakeStore = {
  dispatch: jest.fn(),
  getState: () => ({
    produces: initialProducesState,
  }),
  subscribe: jest.fn()
};

describe('TrendsFilters', () => {

  const props = {
    data: {
      name: 'Produces',
      items: [],
      allItemsSelected: true,
    },
    selectAllProduces: jest.fn(),
    updateProduce: jest.fn(),
  };

  it('ProfileTemplate snapshot"', () => {
    const wrapper = mount(
      <Provider store={fakeStore}>
        <TrendsFilters {...props} />
      </Provider>
    );

    const component = renderer.create(wrapper)
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  });

  

})