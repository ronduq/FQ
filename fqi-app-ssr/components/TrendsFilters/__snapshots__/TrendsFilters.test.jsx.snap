// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrendsFilters ProfileTemplate snapshot" 1`] = `
Array [
  <div
    className="filterTitle"
  >
    Filters
  </div>,
  <fieldset
    className="container"
  >
    <legend>
      Compare by
    </legend>
    <div
      className="radio"
    >
      <input
        autoComplete="off"
        checked={false}
        id="quality"
        name="radio1"
        onChange={[Function]}
        tabIndex="0"
        type="radio"
        value="quality"
      />
      <label
        htmlFor="quality"
      >
        Quality
      </label>
    </div>
    <div
      className="radio"
    >
      <input
        autoComplete="off"
        checked={false}
        id="value"
        name="radio1"
        onChange={[Function]}
        tabIndex="0"
        type="radio"
        value="value"
      />
      <label
        htmlFor="value"
      >
        Value
      </label>
    </div>
  </fieldset>,
  <div
    className="filterContainer"
  >
    <fieldset>
      <button
        aria-expanded={true}
        className="header"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="titleIcon iconMedium iconTeal"
          focusable="false"
        >
          <use
            xlinkHref="#Produces"
          />
        </svg>
        <legend
          className="title"
        >
          Produces
        </legend>
        <svg
          aria-hidden="true"
          className="arrowIcon iconSmall iconOrange arrowDown"
          focusable="false"
        >
          <use
            xlinkHref="#chevron"
          />
        </svg>
      </button>
      <ul
        className="list"
      >
        <li
          className="listItem listItemMain"
        >
          <div
            className="checkbox"
          >
            <input
              autoComplete="off"
              checked={true}
              id="AllProduces"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              aria-hidden="true"
              htmlFor="AllProduces"
            >
              <span>
                All
                 (0)
              </span>
            </label>
          </div>
        </li>
      </ul>
    </fieldset>
  </div>,
]
`;
